src/contents/all/components/knot/Knot/Knot.tsx

```diff
@@ -20,37 +23,82 @@ import '../Trix.scss';
 //   </div>
 // </div>
 
-const Knot = props => (
-  <Timeline.Item className={`knot ${props.isLast ? "ant-timeline-item-last" : ""}`}>
-    <div className={`knotCard ${props.knot.private ? "private" : ""}`}>
-      <div className="trix-content" dangerouslySetInnerHTML={{__html: props.knot.body}} />
-    </div>
-    <div className="knotMeta">
-      <Moment
-        filter={(value) =>
-          value.replace(/^a few seconds ago/g, 'just now')
-          .replace(/^a /g, '1 ')
-          .replace(/^an /g, '1 ')
-          .replace("minute", 'min')
-        }
-        fromNow
-      >
-        {props.knot.postedAt}
-      </Moment>
-      <Popconfirm
-        title="Delete note?"
-        okText="Delete"
-        okType="default"
-        overlayClassName="deleteConfirm"
-        icon={null}
-        onConfirm={deleteKnot}
-      >
-        <span className={`hasAction ${noteHover || isMobile ? "show" : "hide"}`}>
-          {deleteLoading ? <Icon type='loading' /> : 'Delete'}
-        </span>
-      </Popconfirm>
-    </div>
-  </Timeline.Item>
-)
+// <span className={`hasAction ${noteHover || isMobile ? "show" : "hide"}`}>
+//   {deleteLoading ? <Icon type='loading' /> : 'Delete'}
+// </span>
+
+interface Props {
+  log: Log
+  key: number
+  knot: Knot
+  index: number
+  totalKnots: number
+  isLast: boolean
+  createKnot: (bodyHtml: string, bodyText: string) => void
+}
+
+// <div className={`knotCard ${props.knot.private ? "private" : ""}`}>
+//   <div className="trix-content" dangerouslySetInnerHTML={{__html: props.knot.body}} />
+// </div>
+
+const Knot = props => {
+
+  const [loading, setLoading] = useState(false)
+
+  const editKnot = (bodyHtml, bodyText) => {
+    setLoading(true)
+    chrome.runtime.sendMessage({
+      message: "edit_knot",
+      log: props.log,
+      knot: {
+        body: bodyHtml,
+        bodyText: bodyText
+      }
+    })
+  }
+
+  const deleteKnot = (knot) => {
+    setLoading(true)
+    chrome.runtime.sendMessage({
+      message: "delete_knot",
+      log: props.log,
+      knot: props.knot
+    })
+  }
+
+  return (
+    
+    <Timeline.Item className={`knot ${props.isLast ? "ant-timeline-item-last" : ""}`}>
+      <KnotInput
```

Comment:
So now an instance of KnotInput replaces the knotcard from before, passing an existing knot and editKnot function (rendering this is causing the extension to crash i believe...)

Reply:
do i maybe have to tweak the scss?

Reply:
hmm would look into the logs in dev console on the webpage you're on itself to see what errors the extension is giving you that is making it crash.